sale_purchase_force_vendor/models/sale_order_line.py

```diff
@@ -18,17 +20,36 @@ class SaleOrderLine(models.Model):
         comodel_name="res.partner",
         string="Vendor",
     )
+    vendor_id_domain = fields.Char(
+        compute="_compute_vendor_id_domain",
+        readonly=True,
+        store=False,
+    )
     allowed_vendor_ids = fields.One2many(
         comodel_name="res.partner",
         string="Proveedor",
         compute="_compute_allowed_vendor_ids",
     )
 
+    @api.depends("allowed_vendor_ids")
+    def _compute_vendor_id_domain(self):
+        for item in self:
+            domain = (
+                [("id", "in", item.allowed_vendor_ids.ids)]
+                if item.order_id.sale_purchase_force_vendor_restrict
+                else [(1, "=", 1)]
+            )
+            item.vendor_id_domain = json.dumps(domain)
+
     @api.depends("product_id")
     def _compute_allowed_vendor_ids(self):
```

Comment:
Why keeping this field?

Reply:
You are right, `allowed_vendor_ids` field is no longer needed, it is already changed.